### DISCUSSION TITLE
Any plans to support SK in Java?

### DISCUSSION THREAD

**Discussion by ocuenca-coursera** [microsoft/semantic-kernel#381] [General]
I saw Microsoft is focussing in develop SK in .NET and Python. Any plans for Java?

**Comment by timlaverty** [comment]
Hi, we are looking at additional language support including Java.  Python will be in main soon and we have Typescript in our experimental branch.  One question I'd ask is whether you could use from Java either a Python or C# kernel behind a web service (e.g., behind Azure Functions).  Have you considered this kind of architecture?  

I'd love to understand the drive to a Java specific kernel.

**Reply by ocuenca-coursera** [reply]
Our stack is based on Java, we are looking a framework that can help us to create and test multi model solutions where everything would be decoupled and easy to change. You are right though that maybe using lambdas in AWS or Functions in Azure could be an alternative solution, but I would rather to use and even participate in the development of this new framework directly in java, to have the possibility to add new features when is needed or adapt to specific use cases. This is evolving really fast and I would like to have those options

**Comment by timlaverty** [comment]
Great feedback, thanks.  We will loop back with more concrete plans.

**Reply by ocuenca-coursera** [reply]
Thanks so much Tim, I really appreciate that. Let me know if you have any other question. I will be happy to participate in the java version

**Comment by brunoborges** [comment]
> I saw Microsoft is focussing in develop SK in .NET and Python. Any plans for Java?

Hey!

It's officially released: https://devblogs.microsoft.com/semantic-kernel/introducing-semantic-kernel-for-java/
